vendor/k8s.io/apimachinery/go.mod

```diff
@@ -2,13 +2,13 @@
 
 module k8s.io/apimachinery
 
-go 1.13
+go 1.15
 
 require (
 	github.com/davecgh/go-spew v1.1.1
 	github.com/docker/spdystream v0.0.0-20160310174837-449fdfce4d96
 	github.com/elazarl/goproxy v0.0.0-20180725130230-947c36da3153
-	github.com/evanphx/json-patch v4.2.0+incompatible
+	github.com/evanphx/json-patch v0.0.0-20190815234213-e83c0a1c26c8
```

Comment:
Perhaps something for @dims - I noticed kubernetes switched to this version to get a fix/feature  in. This commit is somewhere between v4.5.0 and v4.6.0:
https://github.com/evanphx/json-patch/compare/e83c0a1c26c884f8ec4fb743f6db92a941f605f1...v4.6.0

There's also a v4.7.0 release, that fixes some bugs:
https://github.com/evanphx/json-patch/compare/v4.6.0...v4.7.0


So, perhaps k/k can update to that version to get back on a released version of the dependency. I can try opening a PR to k/k (not sure if there's special magic needed to update dependencies there 😅), or perhaps someone else has time to do so

Reply:
@thaJeztah we are a bit sensitive to json-patch historically i think. Will run it by @liggitt. It is ok for containerd to switch first. We will just pick it up when we rev to containerd 1.4.0

Reply:
between v4.5.0 and v4.6.0, json-patch did the following things in this order:

https://github.com/evanphx/json-patch/compare/v4.5.0...v4.6.0

1. added a go.mod file which made semver revisions unaddressable until they renamed to `github.com/evanphx/json-patch/v5` (done in v5.0.0)
2. fixed an issue with significant performance implications
3. changed behavior to be RFC-compliant, but which is potentially breaking for API consumers

We are not yet comfortable picking up the behavior-breaking changes (especially not to backport to a patch release, see discussion in https://github.com/kubernetes/kubernetes/pull/91622#issuecomment-642781591), but did want the bugfix.

Reply:
> It is ok for containerd to switch first. We will just pick it up when we rev to containerd 1.4.0

Heh, well, containerd (nor containerd/cri) currently vendor it / depend on it (it was actually a facepalm moment that I looked at the versions, then realised it's not used here) 😂 

> changed behavior to be RFC-compliant, but which is potentially breaking for API consumers

> We are not yet comfortable picking up the behavior-breaking changes (especially not to backport to a patch release, see discussion in https://github.com/kubernetes/kubernetes/pull/91622#issuecomment-642781591, but did want the bugfix.

Ahm, gotcha, I thought it was updated to get the fix, and this commit was "latest" at that point, and it wasn't updated after.

> We are not yet comfortable picking up the behavior-breaking changes

That makes perfect sense. So looking at the diff, it looks like https://github.com/evanphx/json-patch/commit/a5df74aab0efb032ca893c048330abcad0b78e6f is the offending commit. This actually sounds like the exact scenario for which go modules were designed. Given that this is a (potentially) breaking change, that warrants a major version bump, so it could (/should) be considered a mistake that it was merged in a minor release.

Perhaps we should suggest a revert of that commit, and a v4.8.0 tag to be created. Given that they have a v5.x version now, this gives consumers of the package the opportunity to opt-in to the new behavior when ready, and others to stay on v4.x until they're ready for the change.

I guess Kubernetes is the most prominent consumer of that package (https://grep.app/search?q=github.com/evanphx/json-patch), and still is on v4 (kubernetes-sigs/kind looks to have made the switch to v5, so should not be affected by a revert https://grep.app/search?q=github.com/evanphx/json-patch/v5)

Reply:
I went ahead and opened a pull request for consideration; https://github.com/evanphx/json-patch/pull/111

Reply:
The tagged v4.8.0 is functionally correct with no breaking changes, but still contains a go.mod file requiring it to be addressed as `v0.0.0-20200808040245-162e5629780b` in go.mod files

Opened https://github.com/evanphx/json-patch/pull/113 to drop the go.mod file in the 4.x stream for tidyness, though it would only affect usage by go module-based consumers.

Reply:
btw, json-patch v4.9.0 was just released that once again allows addressing as `v4.9.0+incompatible` via go modules

Reply:
Perfect! Thanks for the update